### ISSUE TITLE
Missing display of formatted citation (and more) for OSM import

### ISSUE THREAD

**Issue by serviliusahala** [isawnyu/pleiades-gazetteer#231]
It seems that the formatted citation generated by the OSM import tool is now not displaying on nodes, ways imported from OSM. Upon opening the location in edit mode, the url appears in the 'other identifier' field, but Formatted citation is blank. An example here - https://pleiades.stoa.org/places/766392/osm-location-of-ras-muhammad
![screenshot 2016-10-06 22 04 36](https://cloud.githubusercontent.com/assets/3477629/19176758/2f68d396-8c11-11e6-9af2-823c2e4bc911.png)

**Comment by paregorios** [comment]
Note that this ticket should be resolved once [Scrumdo story 96](https://www.scrumdo.com/projects/story_permalink/1103944) is implemented.

**Comment by paregorios** [comment]
OSM import needs to be updated to construct new-style references. 

## Adjustments to the OSM import code

The current implementation has the following needs:
- OSM import seems to be putting the URI in the "Other identifier" field instead of in the "Access URI" field.
- OSM import is not constructing a "formatted citation". It should construct formatted citations of the form: "osm:{type}={osmid}" where "{type}" is one of "node", "way", or "relation" and where "{osmid}" is the numeric identifier for the source data item in OpenStreetMap.
- OSM import is not constructing a "short title". It should insert "OSM" for the short title.
- OSM import is not constructing a "citation detail." It should construct a citation detail like "{Type} {osmid}" where "{Type}" is one of "Node", "Way", or "Relation" (with the first and only first letter of the word capitalized) and where "{osmid}" is as above for formatted citation.

## Content Migration

Existing locations deriving from OSM content will need migration (e.g., https://pleiades.stoa.org/places/462077/monte-etna). Those created prior to the update to our "references" will probably have proper "formatted citations" and "access uris" because of the generalized migration we did; however, they will still need "short title" and "citation detail" as described above. 

Locations imported from OSM since the implementation of our new-style relations (e.g., https://pleiades.stoa.org/places/766392/osm-location-of-ras-muhammad) will need all of the concerns above addressed in migration (i.e., "short title", "citation detail", "formatted citation", "access uri", and will need the access URI value _removed_ from the "other identifier" field.

**Comment by davisagli** [comment]
I have fixed this and ran the migration on staging.

**Comment by davisagli** [comment]
Here is the migration script if you'd like to skim and make sure it looks sane:

```
import transaction

catalog = app.plone.portal_catalog
i = 0
for b in catalog.unrestrictedSearchResults(portal_type='Location'):
    ob = b._unrestrictedGetObject()
    citations = ob.getReferenceCitations()

    changed = False
    for citation in citations:
        uri = None
        if 'http://www.openstreetmap.org' in citation['identifier']:
            citation['access_uri'] = uri = citation['identifier']
            citation['identifier'] = ''
        elif 'http://www.openstreetmap.org' in citation['access_uri']:
            uri = citation['access_uri']
        if uri is not None:
            parts = uri.strip('/').split('/')
            objtype, objid = parts[-2:]
            if objtype not in ('node', 'way', 'relation'):
                continue
            if not citation['formatted_citation']:
                citation['formatted_citation'] = 'osm:{}={}'.format(objtype, objid)
            if not citation['short_title']:
                citation['short_title'] = 'OSM'
            if not citation['citation_detail']:
                citation['citation_detail'] = '{} {}'.format(objtype.title(), objid)
            changed = True
    if changed:
        ob.setReferenceCitations(citations)
        print b.getPath()
    i += 1
    if not i % 100:
        print i
        transaction.commit()

transaction.commit()
```

**Comment by paregorios** [comment]
@davisagli Import works on staging. Migrated content on staging looks good. Please deploy to production and run migration at whatever time seems best to you.

**Comment by davisagli** [comment]
Okay, all set on production now too.

**Comment by paregorios** [comment]
My test of the OSM import on production worked, and migrated OSM locations I checked look ok as well. @serviliusahala would you please have a look at this as well and let us know if it looks good to you? If so, we'll close the ticket as resolved.

**Comment by serviliusahala** [comment]
@paregorios asked me to check back on this issue. As a test I imported an OSM location (https://pleiades.stoa.org/places/658564/osm-location-of-bagras-castle/). The import seems to have behaved properly and, after import, the Data Source, with link, is displayed clearly as OSM Way 219558575. From this limited test, it would seem that the earlier behavior where the OSM reference was included but not outwardly visible has been resolved.
